plugins/filesystem/dropbox/Adapter/JoomlaDropboxAdapter.php

```diff
@@ -0,0 +1,388 @@
+	 * @throws  \Exception
+	 */
+	public function getFiles($path = '/', $filter = '')
+	{
+		if ($path != '/' && !$this->dropbox->has($path))
+		{
+			throw new FileNotFoundException("File not found");
+		}
+
+		$response = $this->client->listFolder($path);
+		$files = [];
+
+		foreach ($response['entries'] as $fileEntry)
+		{
+			$files[] = $this->getFileInfo($fileEntry);
+		}
+
+		return $files;
+	}
+
+	/**
+	 * Extract file information from an entry of dropbox
+	 *
+	 * @param   array  $fileEntry  File entry from dropbox
+	 *
+	 * @return \stdClass
+	 * @since   __DEPLOY_VERSION__
+	 */
+	private function getFileInfo($fileEntry)
+	{
+		$file                          = new \stdClass;
+		$file->type                    = ($fileEntry['.tag'] == 'file' ? 'file' : 'dir');
+		$file->name                    = $fileEntry['name'];
+		$file->path                    = $fileEntry['path_display'];
+		$file->size                    = 0;
+		$file->width                   = 0;
+		$file->height                  = 0;
+		$file->create_date_formatted   = '';
+		$file->modified_date_formatted = '';
+		$file->create_date             = '';
+		$file->modified_date           = '';
+		$file->extension               = '';
+		$file->thumb_path              = '';
+
+		// Dropbox does not support Mime Types
+		$file->mime_type               = '';
+
+		if (isset($fileEntry['size']))
+		{
+			$file->size = $fileEntry['size'];
+		}
+
+		if (isset($fileEntry['client_modified']))
+		{
+			$file->create_date_formatted = $fileEntry['client_modified'];
+			$file->create_date = $fileEntry['client_modified'];
+		}
+
+		if (isset($fileEntry['server_modified']))
+		{
+			$file->modified_date_formatted = $fileEntry['server_modified'];
+			$file->modified_date = $fileEntry['server_modified'];
+		}
+
+		if (isset($fileEntry['media_info']))
+		{
+			$mediaInfo = $fileEntry['media_info'];
+			if (isset($mediaInfo['metadata']))
+			{
+				$metaData = $mediaInfo['metadata'];
+				if (isset($metaData['dimensions']))
+				{
+					$dimensions   = $metaData['dimensions'];
+					$file->width  = $dimensions['width'];
+					$file->height = $dimensions['height'];
+				}
+			}
+		}
+
+		if ($file->type == 'file')
+		{
+			$file->extension = substr(strrchr($file->name,'.'),1);
+		}
+
+		if (in_array($file->extension, $this->supportedThumbnailImageFormats))
+		{
+			$file->thumb_path = $this->getThumbnailUrl($fileEntry['id'], $file->modified_date_formatted , $file->path);
+		}
+
+		return $file;
+	}
+
+	/**
+	 * @param   string  $id            File ID provided by dropbox
+	 * @param   string  $timeModified  Time modified
+	 * @param   string  $path          Path to file
+	 *
+	 * @return string
+	 * @since   __DEPLOY_VERSION__
+	 */
+	private function getThumbnailUrl($id, $timeModified , $path)
+	{
+
+		$name = explode(":", $id)[1];
+		$timeStamp = strtotime($timeModified);
+		$filePath = \JPath::clean(JPATH_PLUGINS . '/filesystem/dropbox/.thumb_cache/' . $name . $timeStamp . '.jpg' , '/');
```

Comment:
you are suggesting me to create a folder in media/plg_filesystem_dropbox ? and refer that as the cache right?

Since this was a pure demo i thought to keep all the files within plugin folder, it will be easy to remove it when its not needed

(We don't ship this plugin I assume)